HW2/P2/P2.py

```diff
@@ -74,9 +77,19 @@
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
-    N = 10
+    Nlistc = []
     counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained correlated: {} seconds".format(t.interval))
+    for N in range(1,21):
+		with Timer() as t:
+			move_data_medium_grained(counts, src, dest, 100, N)
+		assert counts.sum() == total, "Wrong total after move_data_medium_grained"
+		Nlistc.append(t.interval)
+		print("N: {}  Medium grained correlated: {} seconds".format(N, t.interval))
+		
+plt.plot(np.arange(1,21), Nlist, color='b', label='Uncorrelated')
```

Comment:
This code is very clean and looks great; the only change I would suggest are more descriptive variable names for Nlistc and Nlist